### PR TITLE
[@types/redux-seamless-immutable] Add definitions

### PR DESCRIPTION
**Pull request by SoaresMG**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

Package: [redux-seamless-immutable](https://github.com/eadmundo/redux-seamless-immutable)

### CODE REVIEW

File: types/redux-seamless-immutable/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// Type definitions for 0.4.0
+// Project: https://github.com/eadmundo/redux-seamless-immutable
+// Definitions by: Leandro Soares <https://github.com/SoaresMG>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.3
+
+declare module "redux-seamless-immutable" {
```
Comment (sheetalkamat) [inline types/redux-seamless-immutable/index.d.ts]: Do not declare module here. Just take everything here outside the module declaration.

File: types/redux-seamless-immutable/tslint.json [hunk 2]
```diff
@@ -0,0 +1,79 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
```
Comment (sheetalkamat) [inline types/redux-seamless-immutable/tslint.json]: New package cannot override any rules. Use only extends rule and remove everything else and fix errors if any.

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/redux-seamless-immutable/index.d.ts (hunk 1 above)
Comment (SoaresMG) [reply types/redux-seamless-immutable/index.d.ts]: Btw how does it work if there are no modules declared? I saw some packages doing it.

Comment (sheetalkamat) [reply types/redux-seamless-immutable/index.d.ts]: Your `index.d.ts` is defining shape of `redux-seamless-immutable` module

**Review by sheetalkamat — APPROVED** [review]

**Pull request merged**